src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -720,6 +720,14 @@ public function getSession()
         $session = $this->session;
         if (!$session instanceof SessionInterface && null !== $session) {
             $this->setSession($session = $session());
+            /*
```

Comment:
Why shouldn't this be running if `$session instanceof SessionInterface`?

Should we move this new block outside of the current if-statement and into a new one with: 

```php
if ($session instanceof SessionInterface) {
    if (!$session->isStarted()) {
      $sessionId = $this->cookies->get($session->getName(), '');
      $session->setId($sessionId);
    }
}
```

Reply:
Its only required to set once when the Session is beinged created by the SessionFactory is being created and the `$session()` creating the Session here. Another place could be doing it directly in the SessionFactory.

Reply:
Hm.. Is that cleaner? 
To modify the callable in `AbstractSessionListener::onKernelRequest()`?

Reply:
Sounds like a good idea so also the handling is done all in the HttpKernel component.